cassandra-plugins/src/main/java/io/cdap/plugin/batch/source/BatchCassandraSource.java

```diff
@@ -273,6 +272,22 @@ private Schema getSchema() {
         throw new IllegalArgumentException("Invalid schema: " + e.getMessage());
       }
     }
+
+    public void validate(FailureCollector collector) {
+      if (Strings.isNullOrEmpty(schema)) {
+        collector.addFailure("Schema must be specified.", null).withConfigProperty(Cassandra.SCHEMA);
+      }
+      if (port != null && (port < 1 || port > 65535)) {
+        collector.addFailure("Invalid port.", "Port number must be an integer from 1 to 65535.")
+          .withConfigProperty(BatchCassandraSink.Cassandra.PORT);
+      }
+      if ((!containsMacro(Cassandra.USERNAME) && !containsMacro(Cassandra.PASSWORD)) &&
```

Comment:
more clear to put in this way:

!containsMacro(Cassandra.USERNAME) && !containsMacro(Cassandra.PASSWORD) && (Strings.isNullOrEmpty(username) ^ Strings.isNullOrEmpty(password))

Reply:
Fixed